### ISSUE TITLE
Consider renaming pad_batch to stack_and_pad

### ISSUE THREAD

**Issue by PetrochukM** [PetrochukM/PyTorch-NLP#59]
"pad batch" is not very informative, as if your padding an existing pad, while "stack and pad" follows along the lines of the ``torch.stack`` module naming.

**Comment by PetrochukM** [comment]
Done by https://github.com/PetrochukM/PyTorch-NLP/pull/68
